### ISSUE TITLE
Improve 'firecracker --version`

### ISSUE THREAD

**Issue by rn** [firecracker-microvm/firecracker#745]
`firecracker` built from HEAD of master (commit `497b4fe358c5`) shows:
```
% ./firecracker --version
firecracker 0.11.0
```
This is clearly a few commits ahead of the 0.11.0 release.

It would be better if it reports something like `0.11.0+ <git hash>` to easier identify the version (note the `+`), or just the commit hash for none releases . If `firecracker` was build from a dirty repository (ie local, un-committed changes) indicating this with `dirty` would also help.

**Comment by aghecenco** [comment]
Would the output of `git describe` do the trick?

Release commit 3ad0b7b6284853f3b38d1ec7c2d81e1fccfcddfd :
```bash
git describe 3ad0b7b6284853f3b38d1ec7c2d81e1fccfcddfd
v0.12.0
```

Following commit d9c0086bda99e5ec183eb80feccee2f158647fea :
```bash
git describe d9c0086bda99e5ec183eb80feccee2f158647fea
v0.12.0-1-gd9c0086
```

**Comment by rn** [comment]
yeah, that looks good. I would maybe also add a `-dirty` if build from a repository which does not have uncommitted changes. The reliable way to get this is with something like this:
```
git update-index -q --refresh && git diff-index --quiet HEAD -- . || echo "-dirty"
```
[the `update-index` is necessary in case the timestamp (but no the content) of a local file changed]

**Comment by 61131** [comment]
Within another code base, I use the following for generating a short description for version reporting:

`GIT_VERSION:= $(shell git --no-pager describe --tags --always --dirty)`

For a current checkout, it reports the version as:

`v0.12.0-26-g608aa56`

**Comment by aghecenco** [comment]
@rn I'm confused, are repos **without** uncommitted changes dirty? I was thinking the opposite.

**Comment by Jibbow** [comment]
There is a library which does exactly that: https://docs.rs/git-version/0.3.2/git_version/index.html
However, quickly pulling it into Firecracker gives me a `No such file or directory` error inside the macro. Should I investigate this error further or is this the wrong way to go anyways because this adds a new dependency?

**Comment by aghecenco** [comment]
See also https://github.com/cloud-hypervisor/cloud-hypervisor/pull/746/commits/24195b01026890696a5902543076ba42617c6ae3

**Comment by vbmade2000** [comment]
May be https://docs.rs/built/0.4.2/built/ cab help ?

**Comment by alindima** [comment]
Issue was fixed by PR #2061
